### DISCUSSION TITLE
~ jim - Publishing content in Quarto using conditionals and if/else code flow

### DISCUSSION THREAD

**Discussion by giscus** [jimgar/jimgar.github.io#7] [General]
# ~ jim - Publishing content in Quarto using conditionals and if/else code flow

A selection of simple methods using the knitr engine and HTML output

https://jimgar.github.io/posts/quarto-code-conditionals/post.html

**Comment by lamurian** [comment]
Hey Jim, nice article! You might have found another workaround to make `{{< include >}}` conditional. Just leaving my two cents here, I usually wrap `{{< include >}}` in a conditional div, e.g.:

```
::: {.content-visible when-meta="include.this"}
{{< include this.qmd >}}
:::
```

Where `when-meta` (or `unless-meta`) corresponds to the metadata in your `YAML`. A complete minimal working example would be as follow:

File: `main.qmd`

```markdown
---
include:
  this: true
---

This is the main markdown, and I want to include other files conditionally.

::: {.content-visible when-meta="include.this"}
{{< include this.qmd >}}
:::

```

File: `this.qmd`

```markdown
This is the file I included.
```

I often use this in my writing, you can see it in action: https://github.com/namakala/medication-trend/blob/6a66b4aaf2857852c377e5925871eeaa7ecdc03f/docs/index.qmd#L9-L11
